Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs

```diff
@@ -178,33 +179,57 @@ public ISiteBuilder CreateBuilder(ITracer tracer, ILogger logger, IDeploymentSet
 
         private ISiteBuilder ResolveNonAspProject(string repositoryRoot, string projectPath, IDeploymentSettingsManager perDeploymentSettings)
         {
+            string strFramework = System.Environment.GetEnvironmentVariable(OryxBuildConstants.OryxEnvVars.FrameworkSetting);
+            Framework framework = SupportedFrameworks.ParseLanguage(strFramework);
+
             string sourceProjectPath = projectPath ?? repositoryRoot;
-            if (FunctionAppHelper.LooksLikeFunctionApp())
-            {
-                return new FunctionBasicBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsNodeSite(sourceProjectPath))
-            {
-                return new NodeSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsPythonSite(sourceProjectPath))
-            {
-                return new PythonSiteBuilder(_environment, perDeploymentSettings, _propertyProvider, repositoryRoot, projectPath);
-            }
-            else if (IsGoSite(sourceProjectPath))
+            switch (framework)
```

Comment:
@sanchitmehta It shouldnt be possible to the a blessed image for the linuxfxversion, and the customer using the sniffer to detect another framework,  The sniffed runtime would not be present.